src/vaadin-sub-menu-mixin.html

```diff
@@ -0,0 +1,137 @@
+<!--
+@license
+Copyright (c) 2019 Vaadin Ltd.
+This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+-->
+
+<script>
+  /**
+   * @namespace Vaadin
+   */
+  window.Vaadin = window.Vaadin || {};
+
+  /**
+   * @namespace Vaadin.Menu
+   */
+  window.Vaadin.Menu = window.Vaadin.Menu || {};
+
+  /**
+   * @polymerMixin
+   */
+  Vaadin.Menu.SubMenuMixin = superClass => class SubMenuMixin extends superClass {
+
+    constructor() {
+      super();
+      this.__boundOutsideClickListener = this.__outsideClickListener.bind(this);
+      this.__boundOnContextMenuKeydown = this.__onContextMenuKeydown.bind(this);
+    }
+
+    static get observers() {
+      return [
+        '_itemsChanged(items, items.splices)'
+      ];
+    }
+
+    ready() {
+      super.ready();
+
+      this.shadowRoot.addEventListener('button-click', this.__onButtonClick.bind(this));
+
+      this._subMenu.addEventListener('item-selected', this.__onItemSelected.bind(this));
+
+      const overlay = this._subMenu.$.overlay;
+      overlay.modeless = true;
+      overlay.addEventListener('keydown', this.__boundOnContextMenuKeydown);
+      overlay.addEventListener('vaadin-overlay-open', this.__alignOverlayPosition.bind(this));
+    }
+
+    connectedCallback() {
+      super.connectedCallback();
+      document.addEventListener('click', this.__boundOutsideClickListener);
```

Comment:
Oh, this should probably be listened in capture phase. Now it doesn't invoke if the click event's propagation is stopped (for example by another menu's buttons).

Reply:
Good catch, will update. Not sure if I should change any tests to detect that, but will look how this is tested in context-menu itself and possibly overlay as well.